nucypher/characters/lawful.py

```diff
@@ -606,6 +606,124 @@ def __bytes__(self):
+                      host: str,
+                      port: int,
+                      certificate_filepath,
+                      federated_only: bool = False,
+                      *args, **kwargs
+                      ):
+
+        response = network_middleware.node_information(host, port, certificate_filepath=certificate_filepath)
+        if not response.status_code == 200:
+            raise RuntimeError("Got a bad response: {}".format(response))
+
+        stranger_ursula_from_public_keys = cls.from_bytes(response.content, federated_only=federated_only, *args, **kwargs)
+        return stranger_ursula_from_public_keys
+
+
+    @classmethod
+    def from_seednode_metadata(cls,
+                               seednode_metadata,
+                               *args,
+                               **kwargs):
+        """
+        Essentially another deserialization method, but this one doesn't reconstruct a complete
+        node from bytes; instead it's just enough to connect to and verify a node.
+        """
+
+        return cls.from_seed_and_stake_info(checksum_public_address=seednode_metadata.checksum_public_address,
+                                            host=seednode_metadata.rest_host,
+                                            port=seednode_metadata.rest_port,
+                                            *args, **kwargs)
+
+    @classmethod
+    def from_teacher_uri(cls,
+                         federated_only: bool,
+                         teacher_uri: str,
+                         min_stake: int,
+                         ) -> 'Ursula':
+
+        hostname, port, checksum_address = parse_node_uri(uri=teacher_uri)
+        try:
+            teacher = cls.from_seed_and_stake_info(host=hostname,
+                                                   port=port,
+                                                   federated_only=federated_only,
+                                                   checksum_public_address=checksum_address,
+                                                   minimum_stake=min_stake)
+
+        except (socket.gaierror, requests.exceptions.ConnectionError, ConnectionRefusedError):
+            # self.log.warn("Can't connect to seed node.  Will retry.")
+            time.sleep(5)  # TODO: Move this 5
+
+        else:
+            return teacher
+
+    @classmethod
+    @validate_checksum_address
+    def from_seed_and_stake_info(cls,
+                                 host: str,
+                                 port: int,
+                                 federated_only: bool,
+                                 minimum_stake: int = 0,
+                                 checksum_public_address: str = None,
+                                 network_middleware: RestMiddleware = None,
+                                 *args,
+                                 **kwargs
+                                 ) -> 'Ursula':
+
+        #
+        # WARNING: xxx Poison xxx
```

Comment:
Ah? What's going on here?

Reply:
Heh - This started as a note-to-self that I just didn't remove.  Since this is a primary CLI entry point, before we verify the node by certificate, stake or otherwise, *any* URI can be passed here.